Realm/RLMResults.mm

```diff
@@ -376,6 +376,9 @@ - (id)objectAtIndexedSubscript:(NSUInteger)index {
 }
 
 - (id)aggregate:(NSString *)property method:(util::Optional<Mixed> (Results::*)(size_t))method methodName:(NSString *)methodName {
+    if (_results.get_mode() == Results::Mode::Empty) {
```

Comment:
This is incorrect for `sum` (the sum of zero items is zero, not undefined), and it should just be checking for _info being null as that's the thing that actually needs to be guarded against, not calling the aggregate functions on an Empty Results.